src/notifications/notificationsProvider.ts

```diff
@@ -61,12 +63,31 @@ export class NotificationsProvider implements vscode.Disposable {
 		this._notificationsManager.clear();
 	}
 
-	public async getNotifications(sortingMethod: NotificationsSortMethod): Promise<INotificationItem[] | undefined> {
+	public async markAsRead(notification: INotificationItem): Promise<void> {
+		const gitHub = this._getGitHub();
+		if (gitHub === undefined) {
+			return undefined;
+		}
+		await gitHub.octokit.call(gitHub.octokit.api.activity.markThreadAsRead, {
+			thread_id: Number(notification.notification.id)
+		});
+		if (this._notifications === undefined) {
+			return;
+		}
+		const notificationIndexToRemove = this._notifications.findIndex(n => n === notification);
+		if (notificationIndexToRemove !== -1) {
+			this._notifications.splice(notificationIndexToRemove, 1);
```

Comment:
Having a `_notificaitons` member will cause use to be out of sync with the `_notificationsManager`. We should access and manage notifications exclusively through `this._notificationsManager`. I two options:
- Get the notification from the `this._notificationsManager` and set it's `unread` property to `false`. Then, when we [get all notifications in the tree view](https://github.com/microsoft/vscode-pull-request-github/blob/89fe5e32b2cdbe91885c44ba007711806e8ecc5d/src/notifications/notificationsView.ts#L86-L87), we can filter out the onse that have been read.
- Add a `removeNotification` to `NotificationManager`.

Reply:
Right I didn't think about that. I'll make the corresponding change